modules/cf-core/src/machine/xkeys.ts

```diff
@@ -1,66 +1,43 @@
 import { Wallet } from "ethers";
 import { computeAddress, SigningKey } from "ethers/utils";
 import { fromExtendedKey, fromMnemonic, HDNode } from "ethers/utils/hdnode";
+import memoize from "memoizee";
 
 /**
- * Helpful info:
- *
  * BIP-32 specified HD Wallets
  * BIP-39 specifies how to convert mnemonic to/from entropy and mnemonic to seed
  * BIP-43 specifies that the first field should be purpose (i.e. "m / purpose'")
- * BIP-44 specifies that if the purpose is 44, then the format is "m / purpose' / cointype' / account' / change / index"
+ * BIP-44 specifies format of: "m/purpose'/cointype'/account'/change/index" (iff purpose is 44)
  */
 
-export function computeRandomExtendedPrvKey(): string {
-  return fromMnemonic(Wallet.createRandom().mnemonic).extendedKey;
-}
+const xkeyKthHDNode = memoize(
```

Comment:
There's a different memoization package I've seen used in the repo: https://github.com/darrylhodgins/typescript-memoize

Is there a reason you didn't use this one? The decorator syntax is cleaner IMO, but personal preference I guess. If the other one is already a dependency no need to have multiple though.

Reply:
I looked at it & thought it only works for class methods.. looking at it again I'm not sure if that's actually the case. If we can eliminate this new dep & just use the old one that'd def be better though